src/metadata.js

```diff
@@ -0,0 +1,55 @@
+var gcpMetadata = require('gcp-metadata');
+var constants = require('./constants.js');
+
+// prevent self tracing
+var headers = {};
+headers[constants.TRACE_AGENT_REQUEST_HEADER] = 1;
+
+module.exports = function getMetadata(logger, fields, cb) {
+  var result = {};
+  var next = function() {
+    cb(null, result);
+  };
+  for (var i = fields.length - 1; i >= 0; i--) {
+    (function() {
+      var endpoint = fields[i].endpoint;
+      var property = fields[i].property;
+      var onError = fields[i].onError;
+
+      var prevNext = next;
+      next = function() {
+        gcpMetadata[endpoint]({
+          property: property,
+          headers: headers
+        }, function(err, response, body) {
+          if (response && response.statusCode !== 200) {
```

Comment:
wouldn't it make sense to check for err first? Otherwise we end up clobbering the error message.